src/panels/flow-histogram-react/FlowHistogramHelpers.ts

```diff
@@ -1,96 +1,217 @@
-    ]
 }
 
-export const getFlowHistogramPlotConfig = (options: { flowHistogramOptions: FlowHistogramOptionsProps }) => {
-    const axis = options.flowHistogramOptions.direction.label === 'Horizontal' ? 'yaxis' : 'xaxis'
-    let ticks = [[0, 'In'], [1, 'Out']];
-    if (options.flowHistogramOptions.mode.label === 'Stacked') {
-        ticks = [[0, options?.flowHistogramOptions?.display?.label || 'Total']]
+export const getFlowHistogramPlotConfig = (processedData: FlowPanelDataProcessed, options: { flowHistogramOptions: FlowHistogramOptionsProps }) => {
+
+    const stacked = options.flowHistogramOptions.mode.label === 'Stacked'
+    const horizontal = options.flowHistogramOptions.direction.label === 'Horizontal'
+    const container = options.flowHistogramOptions.position.label === 'Under Graph' ? $('.flow-histogram-legend-bottom') : $('.flow-histogram-legend-right') 
+    const noColumns = options.flowHistogramOptions.position.label === 'Under Graph' ? 5 : 1
+    const showLegend = options.flowHistogramOptions.showLegend
+    const legendPosition = options.flowHistogramOptions.position.value 
+    
+    const yaxis = {
+        mode: 'categories',
+        tickLength: 0,
+        ticks: stacked ? processedData.stackedData?.ticks : processedData.separateData?.ticks,
+        autoscaleMargin: 0.02,
     }
-    return {
-        grid: {
-            borderWidth: 0
-        },
+
+    const configOptions: any = {
+        legend: {
+            show: showLegend,
+            container: container,
+            position: legendPosition,
+            backgroundOpacity: 0,
+            noColumns: noColumns,
+            placement: container ? 'outsideGrid' : null,
+            labelFormatter: (label, series) => {return '<a style="margin:3px" >' + label + '</a>'; }
+        },        
         series: {
             bars: {
-                align: 'center',
+                align: "center",
                 barWidth: 0.6,
                 fill: 0.8,
-                horizontal: options?.flowHistogramOptions?.direction?.label === 'Horizontal',
+                horizontal: horizontal,
                 lineWidth: 1,
                 show: true,
-            }
+            },
+            stack: stacked,
         },
-        [axis]: {
-            show: true,
-            autoscaleMargin: 0.02,
-            tickLength: 0,
-            ticks,
-        },
-        legend: {
-            show: options?.flowHistogramOptions.showLegend,
-            position: options?.flowHistogramOptions?.position.value,
-            margin: [-45, -45],
-            backgroundColor: '#181b1f'
+        grid: {
+            borderWidth: 0,
         },
+        xaxis: horizontal ? {} : yaxis,
+        yaxis: horizontal ? yaxis : {}
     }
+    return configOptions
+
 }
+
+export const getLabeledValues = (data, options: { flowHistogramOptions: FlowHistogramOptionsProps }): FlowPanelDataProcessed => {
+
+    let labeledValues: FlowPanelDataProcessed = {}
```

Comment:
This function needs to be broken into more chunks. ~100 line methods are tough to maintain/update/upgrade, and unless there's a really good reason for the length I think we should shorten this one down.